wordpress-mobile/WordPress-iOS
# Jetpack App: New Target, No App Extensions

## Description

This PR adds a new Jetpack target and scheme, so we can build and run a Jetpack app 🚀 

After investigating several different approaches to building a new Jetpack app, we decided that a target / scheme approach was the most feasible solution.

## Changes

- Added a new **Jetpack target and scheme**
- Removed all **App Extensions** from the Jetpack target
- Updated the **Podfile** to include the Jetpack target
- Added **Jetpack icons and launch screens**
- Added a **Jetpack flag** to preprocessor macros

## Screenshots

Different app icons | Running both apps | Jetpack launch screen | Jetpack preprocessor flag 
--- | --- | --- | ---
![Simulator Screen Shot - iPhone 12 Pro - 2021-03-18 at 15 25 11](https://user-images.githubusercontent.com/6711616/111582916-e5b1a680-87fe-11eb-8141-77226db1587b.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-03-18 at 15 25 44](https://user-images.githubusercontent.com/6711616/111582974-011cb180-87ff-11eb-8c1f-c8ce51dff904.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-03-18 at 15 25 14](https://user-images.githubusercontent.com/6711616/111582995-0aa61980-87ff-11eb-913c-b6f305cb0de8.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-03-18 at 15 25 23](https://user-images.githubusercontent.com/6711616/111583008-109bfa80-87ff-11eb-9a1c-08c1e3000b3c.png)

## How to test

1. Run `bundle exec pod install`
2. Build the WordPress scheme
3. ✅ Should build as usual
4. Build the Jetpack scheme
5. ✅ Should build and it should have a 1) different icon 2) different splash screen
6. In the Jetpack app, go to My Site > Switch Site
7. ✅ A "Welcome to the Jetpack app!" notice should be printed in the console
8. In the WordPress app, go to My Site > Switch Site
9. ✅ A "Welcome to the Jetpack app!" notice should NOT be printed in the console

## Additional information

* Splash screen and icons are just placeholders and not the final ones

You can find the other explorations here:

* As a new project: https://github.com/wordpress-mobile/WordPress-iOS/pull/16104
* New target + duplicated app extensions: https://github.com/wordpress-mobile/WordPress-iOS/pull/16111
* New target (similar to this one): https://github.com/wordpress-mobile/WordPress-iOS/pull/16095
* We also briefly looking into Tuist and using a script to keep two projects in sync, both alternatives have a high cost
